datasets/jigsaw_toxicity_pred/jigsaw_toxicity_pred.py

```diff
@@ -0,0 +1,128 @@
+
+    VERSION = datasets.Version("1.1.0")
+
+    @property
+    def manual_download_instructions(self):
+        return """\
+            To use jigsaw_toxicity_pred you have to download it manually from Kaggle: https://www.kaggle.com/c/jigsaw-toxic-comment-classification-challenge/data
+            You can manually download the data from it's homepage or use the Kaggle CLI tool (follow the instructions here: https://www.kaggle.com/docs/api)
+            Please extract all files in one folder and then load the dataset with:
+            `datasets.load_dataset('jigsaw_toxicity_pred', data_dir='/path/to/extracted/data/')`"""
+
+    def _info(self):
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=datasets.Features(
+                {
+                    "comment_text": datasets.Value("string"),
+                    "toxic": datasets.Value("float32"),
+                    "severe_toxic": datasets.Value("float32"),
+                    "obscene": datasets.Value("float32"),
+                    "threat": datasets.Value("float32"),
+                    "insult": datasets.Value("float32"),
+                    "identity_hate": datasets.Value("float32"),
+                }
+            ),
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # This method is tasked with downloading/extracting the data and defining the splits depending on the configuration
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        data_dir = os.path.abspath(os.path.expanduser(dl_manager.manual_dir))
+
+        if not os.path.exists(data_dir):
+            raise FileNotFoundError(
+                "{} does not exist. Make sure you insert a manual dir via `datasets.load_dataset('jigsaw_toxicity_pred', data_dir=...)`. Manual download instructions: {}".format(
+                    data_dir, self.manual_download_instructions
+                )
+            )
+
+        test_csv_path = os.path.join(data_dir, "test.csv")
+        test_labels_path = os.path.join(data_dir, "test_labels.csv")
+        output_path = os.path.join(data_dir, "test_expanded.csv")
+
+        df1 = pd.read_csv(test_csv_path)
+        df2 = pd.read_csv(test_labels_path)
+        df3 = df1.merge(df2)
+        df3[df3["toxic"] != -1].to_csv(output_path)
```

Comment:
please don't create new files.

You can load df1 and df2 to build df3 inside _generate_examples, and without creating files.
To get the paths to the csv files in _generate_examples, you can pass the directory where the files for df1 and df2 in `gen_kwargs`.
for example
```
gen_kwargs={"dara_dir": data_dir}
```

`gen_kwargs` is passed as keyword arguments to _generate_examples

this way in _generate examples you can load df1 and df2 since you have their data_dir, or you can load the train data since again it's in the same data_dir